vitrina/helpers.py

```diff
@@ -706,3 +707,98 @@ def validate_file(file):
         owner=None,
         mime_type=mime_type,
     )
+
+
+def build_page_title_context(
+    *,
+    dataset: Dataset | None = None,
+    organization: Organization | None = None,
+    model: StructureModel | None = None,
+    prop: StructureProperty | None = None,
+    language_code: str = "lt",
+) -> dict[str, str]:
+    """Build page title context based on the provided objects."""
+
+    context: dict[str, str] = {}
+
+    if organization is not None:
+        context["title"] = organization.title
+        context["object_type"] = _("Organizacija")
+        return context
+
+    if dataset is None:
+        return context
+
+    _add_common_dataset_context(context, dataset, language_code)
+
+    if prop is not None and model is not None:
+        context["title"] = getattr(prop, "title", None) or prop.name
+        context["object_type"] = _("Modelio laukas")
+    elif model is not None:
+        context["title"] = getattr(model, "title", None) or model.name
+        context["object_type"] = _("Modelis")
+    else:
+        title = dataset.safe_translation_getter("title", language_code=language_code) or ""
+        context["title"] = title
+        context["object_type"] = dataset.subclass.translated_title if dataset.subclass else ""
+
+    return context
+
+
+def _add_common_dataset_context(context: dict[str, str], dataset: Dataset, language_code: str) -> None:
+    """Add common context elements that are shared across dataset-based objects."""
+    try:
+        parent_dataset = _get_parent_dataset(dataset)
+        context["root_label"] = _("Organizacija")
+        context["root_name"] = parent_dataset.organization.title
+    except Exception:
```

Comment:
Any way to specify concrete exceptions instead of most generic one?

Reply:
Same for other places

Reply:
Fixed.